src/Tools/dotnet-monitor/CollectionRules/ActionOptionsDependencyAnalyzer.cs

```diff
@@ -162,6 +182,31 @@ public object SubstituteOptionValues(IDictionary<string, CollectionRuleActionRes
             return settings;
         }
 
+        private bool TryCloneSettings(object originalSettings, ref object settings)
+        {
+            if (originalSettings == null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(originalSettings, settings))
+            {
+                if (originalSettings is BaseRecordOptions baseRecord)
```

Comment:
Could we just make every action settings options type derive from this and not have to check or log a failure (since there should not be a failure)? During configuration registration, we can require that all action settings options types have to implement this base class.

Reply:
I am not opposed to this, but I may make it a separate PR.